### PR TITLE
Port remaining 3.0.4 fixes with adjustments for 3.1.1

### PR DESCRIPTION
**Pull request by handrews**
This PR really has two parts, and you can click each "series of commits" to see the diff of just that part:
1.  A [series of commits](https://github.com/OAI/OpenAPI-Specification/pull/3921/files/9739dfe5afa8cb6fce5f64bbde04eb5f300658bc) that ports, with as little changes as possible, all of the outstanding 3.0.4 changes not yet in 3.1.1
2. A [series of commits](https://github.com/OAI/OpenAPI-Specification/pull/3921/files/9739dfe5afa8cb6fce5f64bbde04eb5f300658bc..31d23da8dc55d13c4cd16e15b07194df376dbc7f) that adjusts those ports for 3.1.1 differences, primarily the change from `format` to `contentEncoding`

***IMPORTANT NOTE:*** The 3.1.0 spec is inconsistent about whether `type: string` _without_ `contentEncoding` and any other primitive type (`number`, `integer`, `boolean`) regardless of `contentEncoding` defaults to a Content Type of `text/plain` (per [§4.8.1.4](https://spec.openapis.org/oas/v3.1.0#special-considerations-for-multipart-content)) or `application/octet-stream` (per the [`contentType` fixed fields entry](https://spec.openapis.org/oas/v3.1.0#fixed-fields-12)).  This PR settles on `text/plain` as that is what the [3.0.3 `contentType` fixed fields entry said](https://spec.openapis.org/oas/v3.0.3#fixed-fields-12), and the change from `format` to `contentEncoding` should not have impacted that.  See also #3922 for a similar fix to 3.0.4.

This is a good opportunity to look for consistency although I will ask that you ***DO NOT*** hold up this PR with consistency/editorial changes that might need to also go into 3.0.4.  Please either open a PR on 3.0.4 with those, or start an issue where we can track them.

I squashed the ported PR commits somewhat, and just retained separate commits when review feedback was substantial enough that I thought it warranted a separate commit for crediting or clarity purposes, or (in the case of 3861) the original PR was a grab-bag of different tiny fixes in the first place.

Ports:
* #3867 (2 consecutive commits)
    * 1st commit includes #3904
* #3864 
* #3818
* #3911 (2 commits, one combined with another PR noted further down this list)
* #3840
* #3841 
* #3894 
* #3861 (11 consecutive commits)
* #3857 (4 consecutive commits)
* #3859 (3 consecutive commits)
    * 3rd commit includes half of #3911) 
* #3917
* #3905 
* #3907 
* #3919 
* #3920

### CODE REVIEW

**Review by lornajane — APPROVED** [review]
An epic, but it seems to be in good shape!

File: versions/3.1.1.md
Comment (ralfhandl) [inline versions/3.1.1.md:154]: The JSON Schema link seems to be for Draft 5, shouldn't it be 
```suggestion
- Tags MUST be limited to those allowed by [YAML's JSON schema ruleset](https://yaml.org/spec/1.2/spec.html#id2803231), which defines a subset of the YAML syntax and is unrelated to [JSON Schema](https://tools.ietf.org/html/draft-bhutton-json-schema-00).
```

File: versions/3.1.1.md
Comment (ralfhandl) [inline versions/3.1.1.md:380]:
```suggestion
<a name="infoVersion"></a>version | `string` | **REQUIRED**. The version of the OpenAPI document (which is distinct from the [OpenAPI Specification version](#oasVersion) or the version of the API being described).
```

File: versions/3.1.1.md
Comment (ralfhandl) [inline versions/3.1.1.md:1781]:
```suggestion
Properties are correlated with `multipart` parts using the `name` parameter to `Content-Disposition: form-data`, and with `application/x-www-form-urlencoded` using the query string parameter names.
```

File: versions/3.1.1.md
Comment (ralfhandl) [inline versions/3.1.1.md:1816]:
```suggestion
<a name="encodingStyle"></a>style | `string` | Describes how a specific property value will be serialized depending on its type.  See [Parameter Object](#parameterObject) for details on the [`style`](#parameterStyle) property. The behavior follows the same values as `query` parameters, including default values. Note that the initial `?` used in query strings is not used in `application/x-www-form-urlencoded` message bodies, and MUST be removed (if using an RFC6570 implementation) or simply not added (if constructing the string manually). This property SHALL be ignored if the request body media type is not `application/x-www-form-urlencoded` or `multipart/form-data`. If a value is explicitly defined, then the value of [`contentType`](#encodingContentType) (implicit or explicit) SHALL be ignored.
```

File: versions/3.1.1.md
Comment (ralfhandl) [inline versions/3.1.1.md:3420]:
```suggestion
* Versions 3.1.0, 3.0.3, and earlier of this specification erroneously used the term "absolute URI" instead of "non-relative URI", so authors using namespaces that include a fragment should check tooling support carefully.
```

**Review by ralfhandl — COMMENTED** [review]
Impressive to see all the accumulated improvements in one PR!

Only minor nits.

**Comment by miqui** [comment]
So far, looks good to me

**Review by miqui — APPROVED** [review]

File: versions/3.1.1.md
Comment (handrews) [reply versions/3.1.1.md:380]: @ralfhandl can you move this over to your typo PR so it goes into 3.0.4 as well?  Then we can port that PR.  After how confusing it was for me to backport merges with that discriminator PR, I'd rather avoid that here and keep everything moving in one direction: 3.0.4 -> 3.1.1 -> 3.2.0

Comment (handrews) [reply versions/3.1.1.md:380]: Or I guess if the 3.04 PR has every change found here in it, we could incorporate those changes here and not merge forward?   (sorry, I should have atleast one cup of tea before commenting on PRs... thinking this through...)

File: versions/3.1.1.md
Comment (handrews) [reply versions/3.1.1.md:1816]: @ralfhandl this also needs adding to your other PR.

**Review by ralfhandl — APPROVED** [review]

File: versions/3.1.1.md
Comment (ralfhandl) [reply versions/3.1.1.md:380]: No worries, let's merge this monster as is (after a re-approval by @lornajane or @miqui) and then merge & forward-port #3924 separately.

Comment (handrews) [reply versions/3.1.1.md:380]: Thanks @ralfhandl - I have a little script that can merge a specified commit across brancheslfiles, so I can run that once your PR is merged if you'd like.

Comment (ralfhandl) [reply versions/3.1.1.md:380]: Yes, please!

Comment (handrews) [reply versions/3.1.1.md:380]: @ralfhandl I have the PR porting the remaining 3.0.4 things ready to go as soon as this PR is merged.

**Comment by handrews** [comment]
This had 3 TSC approvals and it was only a trivial in-PR correction that dismissed the other two - I'm going to merge this to unblock the remaining 3.0.4 ports.

**Pull request merged**
